app/scripts/nav.js

```diff
@@ -27,18 +29,33 @@ $(function() {
 
     lastScrollTop = currentScrollTop;
 
-    if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
-        (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
-      toggleNavOnTop();
+    if (nav.hasClass(onTopClassIndex)) {
+      last_class = onTopClassIndex;
+    } else if (nav.hasClass(onTopClass)){
+      last_class = onTopClass;
+    }
+
+    if (last_class == onTopClass) {
+
+      if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClass) ||
+          (!isOnTop(currentScrollTop) && nav.hasClass(onTopClass)))
+        toggleNavOnTop(onTopClass);
+
+    } else if (last_class == onTopClassIndex) {
+
+      if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClassIndex) ||
+          (!isOnTop(currentScrollTop) && nav.hasClass(onTopClassIndex)))
+        toggleNavOnTop(onTopClassIndex);
+    }
 
   }
```

Comment:
I think you can rewrite this whole block like this:
```
if (last_class === onTopClass || last_class === onTopClassIndex) { 
  toggleNavOnTop(currentScrollTop, last_class);
}
```

and do the verification on `togggleNavOnTop` function, like this:
```
function toggleNavOnTop(currentScrollTop , selected_class) {
  if (isOnTop(currentScrollTop) && !nav.hasClass(onTopClassIndex) ||  (!isOnTop(currentScrollTop) && nav.hasClass(onTopClassIndex)))
    nav.toggleClass(selected_class);
}
```